src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java

```diff
@@ -303,40 +309,39 @@ public String login(@ModelAttribute("authorizeCommand") @Validated AuthorizeRequ
         model.addAttribute(REDIRECT_URI, request.getRedirect_uri());
         model.addAttribute(SCOPE, request.getScope());
         model.addAttribute(SELF_REGISTRATION_ENABLED, request.isSelfRegistrationEnabled());
+
+        final boolean validationErrors = bindingResult.hasErrors();
+        if (validationErrors) {
+            if (StringUtils.isEmpty(request.getUsername())) {
+                model.addAttribute("isUsernameEmpty", true);
+            }
+            if (StringUtils.isEmpty(request.getPassword())) {
+                model.addAttribute("isPasswordEmpty", true);
+            }
+            model.addAttribute(HAS_ERRORS, true);
+            return nextPage;
+        }
+
         try {
-            if (bindingResult.hasErrors()) {
-                if (StringUtils.isEmpty(request.getUsername())) {
-                    model.addAttribute("isUsernameEmpty", true);
-                }
-                if (StringUtils.isEmpty(request.getPassword())) {
-                    model.addAttribute("isPasswordEmpty", true);
-                }
-                model.addAttribute(HAS_ERRORS, true);
+            final String ipAddress = ObjectUtils.defaultIfNull(
+                httpRequest.getHeader(X_FORWARDED_FOR),
+                httpRequest.getRemoteAddr());
+            final String cookie = spiService.authenticate(request.getUsername(), request.getPassword(), ipAddress);
+            final String responseUrl = authoriseUser(cookie, httpRequest);
+            final boolean loginSuccess = responseUrl != null && !responseUrl.contains("error");
+            final boolean policyCheckSuccess = loginSuccess && policyService.evaluatePoliciesForUser(responseUrl, cookie, ipAddress);;
+
+            if (loginSuccess && policyCheckSuccess) {
+                response.addHeader(HttpHeaders.SET_COOKIE, makeCookieSecure(cookie));
+                nextPage = "redirect:" + responseUrl;
+            } else if (loginSuccess && !policyCheckSuccess) {
```

Comment:
I think you don't need the `&& !policyCheckSuccess` here.

Reply:
I don't, added as a suggestion after a feedback from @jburke-amido